src/Entities/Entity.cpp

```diff
@@ -528,9 +528,9 @@ bool cEntity::DoTakeDamage(TakeDamageInfo & a_TDI)
 		Player->GetStatManager().AddValue(statDamageDealt, static_cast<StatValue>(floor(a_TDI.FinalDamage * 10 + 0.5)));
 	}
 
-	m_Health -= static_cast<short>(a_TDI.FinalDamage);
+	m_Health -= static_cast<float>(a_TDI.FinalDamage);
```

Comment:
I think a_TDI.FinalDamage (and RawDamage) should be float as well (and therefore all of that architecture: https://github.com/cuberite/cuberite/blob/master/src/Entities/Entity.cpp#L284), but that's probably outside the scope of this PR (maybe open a new PR if you want to take that on).